nlpserver.py

```diff
@@ -533,6 +548,425 @@ def fasttext():
 
 	return jsonify(data)
 
+@app.route("/image/yolo", methods=['GET', 'POST'])
+def yolo():
+	# Import your Libraries 
+	import torch
+	from torchvision import transforms
+	from PIL import Image, ImageDraw
+	from pathlib import Path
+	from ultralytics import YOLO
+	import pandas as pd
+	import numpy as np
+	from pathlib import Path
+	import requests
+	from io import BytesIO
+	from keras.preprocessing.image import img_to_array
+	from sklearn import preprocessing
+	import json
+
+	intermediate_features = []
+
+	def hook_fn(module, input, output):
+		intermediate_features.append(output)
+
+	def extract_features(intermediate_features, model, img, layer_index=20):##Choose the layer that fit your application
+		hook = model.model.model[layer_index].register_forward_hook(hook_fn)
+		with torch.no_grad():
+			model(img)
+		hook.remove()
+		return intermediate_features[0]  # Access the first element of the list
+	def loadImage(url, size = 640):
+		try:
+			response = requests.get(url)
+			response.raise_for_status()
+		except requests.exceptions.RequestException as err:
+			data['error'] =  err.strerror
+			return jsonify(data)
+	
+		img_bytes = BytesIO(response.content)
+		img = Image.open(img_bytes)
+		img = img.convert('RGB')
+		img = img.resize((size,size), Image.NEAREST)
+		img = img_to_array(img)
+		return img
+	
+	data = dict(default_data)
+	data['message'] = "Yolo -  Parameters: 'iiif_image_url', 'labels' a list of valid labels for object detection (default: face)"
+	data['yolo'] = {}
+	params = {}
+	
+
+	if request.method == 'GET':
+		params['iiif_image_url'] = request.args.get('iiif_image_url')
+		params['labels'] = request.args.getlist('labels')
+		params['norm'] = request.args.get('norm')
+	elif request.method == 'POST':
+		params = request.form # postdata
+	else:
+		data['error'] = 'Invalid request method'
+		return jsonify(data)
+
+	if not params:
+		data['error'] = 'Missing parameters'
+		return jsonify(data)
+
+	if not params['iiif_image_url']:
+		data['error'] = '[iiif_image_url] parameter not found'
+		return jsonify(data)
+
+	if not params['labels']:
+		params['labels'] = ['face']
+
+	if not params['norm']:
+		params['norm'] = 'l2'
+
+	if params['norm'] and params['norm'] not in ['l1','l2','max']:
```

Comment:
l1 not used anywhere? (seeing notes below)

Reply:
Yes not used. Reason I wanted to leave/show that for you bc the comparison for image ML models when using one or the other is semantically so so different. Let's talk about that. I can also use it and then let the SBR processors just call L2.